File: ios/MurphysLaws/Models/ContentPage.swift
//
//  ContentPage.swift
//  MurphysLaws
//
//  Enum representing different content pages in the app
//

import Foundation

public enum ContentPage: String, CaseIterable {
    case about
    case privacy
    case terms
    case contact

    public var title: String {
        switch self {
        case .about:
            return "About"
        case .privacy:
            return "Privacy Policy"
        case .terms:
            return "Terms of Service"
        case .contact:
            return "Contact"
        }
    }

    public var filename: String {
        return "\(self.rawValue).md"
    }
}

File: android/app/src/main/java/com/murphyslaws/domain/model/ContentPage.kt
package com.murphyslaws.domain.model

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
enum class ContentPage(val filename: String, val title: String) : Parcelable {
    ABOUT("about", "About"),
    PRIVACY("privacy", "Privacy Policy"),
    TERMS("terms", "Terms of Service"),
    CONTACT("contact", "Contact");

    val markdownFile: String get() = "$filename.md"
}
